test/functional/combine_logs.py

```diff
@@ -81,13 +81,14 @@ def read_logs(tmp_dir):
 
     # Find out what the folder is called that holds node 0's debug.log file
     debug_logs = list(pathlib.Path(tmp_dir).glob('node0/**/debug.log'))
-    if len(debug_logs) > 0:
-        assert len(debug_logs) < 2, 'Max one debug.log is supported, ' \
-            'found several:\n\t' + '\n\t'.join([str(f) for f in debug_logs])
-        path = debug_logs[0]
-        chain = re.search(r'node0/(.+?)/debug\.log$', path.as_posix()).group(1)  # extract the chain name
-    else:
-        chain = 'regtest'  # fallback to regtest (should only happen when none exists)
+    match len(debug_logs):
+        case 0:
+            chain = 'regtest'  # fallback to regtest
+        case 1:
+            chain = re.search(r'node0/(.+?)/debug\.log$', debug_logs[0].as_posix()).group(1)
+        case _:
+            raise RuntimeError('Max one debug.log is supported, found several:\n\t' +
```

Comment:
Looks like functional test code raises `Exception` and `RuntimeError`. Didn't see a preference in the [developer-notes](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md) or in [test style guidelines](https://github.com/bitcoin/bitcoin/blob/master/test/functional/README.md#style-guidelines), so I'm assuming use of either is ok.

Reply:
Yeah, it's a bit fuzzy to me what error type should be used. Both `Exception` and `RuntimeError` are unfortunately generic. Manually raising `AssertionError` from a "non-assertion-ish"-function felt wrong, also because of the cause being something external to the *combine_logs.py* script itself. Curious if you have any good rationale for such cases.

Reply:
I agree that manually raising `AssertionError` wouldn't have been the way to go (even though it technically was raised in the previous code containing the `assert`).  [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) might have worked, but `RuntimeError` seems fine enough, and seems more appropriate to me than `Exception` (a bit too generic IMO (decent rationale [here](https://stackoverflow.com/a/24065533))).